### PR TITLE
Update script to print external dependencies for a target

### PR DESCRIPTION
**Pull request by dprotaso**
Description:
Allows you to print the external dependencies for a bazel target

Risk Level: 
Low

Testing:
Ran the following bash script
```bash
./tools/print_deps_for_target.sh //source/exe:envoy
```

### CODE REVIEW

File: tools/print_deps_for_target.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import sys
+
+MAP=imp.load_source('deps', 'bazel/target_recipes.bzl')
+
+print('\n'.join([MAP.TARGET_RECIPES[recipe.strip()] for recipe in sys.stdin]))
+" | sort | uniq >> "$repos"
+
+./tools/print_dependencies.py 1> "$all_deps"
+
+cat "$repos" \
```
Comment (htuch) [inline tools/print_deps_for_target.sh]: How do you feel about rewriting this script in Python? Upsides include not needing to install `jq`, less `awk` and bash magic, no need to generate Python code... Downside is you probably have a script that works reasonable already.

Comment (dprotaso) [reply tools/print_deps_for_target.sh]: I thought about that but didn't want to shell out to run `bazel query`

Comment (htuch) [reply tools/print_deps_for_target.sh]: `subprocess.check_output` gives a single line "invoke this shell command, ensure it returns 0 and give me the output as a string".

**Comment by dprotaso** [comment]
Thanks for the suggestion.

I've extended the `tools/print_dependencies.py` script to take a target as an optional argument.

File: tools/print_dependencies.py [hunk 2]
```diff
@@ -33,4 +36,33 @@
...
+  target = sys.argv[1]
+  output = subprocess.check_output(['bazel', 'query', 'deps(%s)' % target])
+
+  repos = set()
+
+  # Gather the explicit list of repositories
+  repo_regex = re.compile('^@(.*)\/\/')
+  for line in output.split('\n'):
+    match = repo_regex.match(line)
+    if match: repos.add(match.group(1))
```
Comment (htuch) [inline tools/print_dependencies.py]: Nit: prefer split lines:
```
if match:
  foo
```
but this is more a personal preference; I haven't seen much Python code like this..

File: tools/print_dependencies.py [hunk 3]
```diff
@@ -33,4 +36,33 @@
...
+  recipe_regex = re.compile('^@envoy_deps//:(\w+)$')
+  for line in output.split('\n'):
+    match = recipe_regex.match(line)
+    if not match: continue
+    key = match.group(1)
+    repo = RECIPE_MAP.TARGET_RECIPES[key]
+    if repo: repos.add(repo)
+
+  deps = filter(lambda dep: dep['identifier'] in repos, deps)
+  print(json.dumps(deps, sort_keys=True, indent=2))
```
Comment (htuch) [inline tools/print_dependencies.py]: Could factor the print/dump into a separate function, so you have consistent formatting across both global and filtered deps.

**Comment by dprotaso** [comment]
@htuch please take a look when you get a chance

**Pull request merged**